app/src/main/java/org/oppia/android/app/drawer/NavigationDrawerHeaderViewModel.kt

```diff
@@ -22,33 +22,41 @@ class NavigationDrawerHeaderViewModel @Inject constructor(
   val profile = ObservableField(Profile.getDefaultInstance())
   private var ongoingTopicCount = DEFAULT_ONGOING_TOPIC_COUNT
   private var completedStoryCount = DEFAULT_COMPLETED_STORY_COUNT
-  val profileProgressText: ObservableField<String> = ObservableField(computeProfileProgressText())
+  val profileTopicProgressText: ObservableField<String> =
+    ObservableField(computeProfileTopicProgressText())
+  val profileStoryProgressText: ObservableField<String> =
+    ObservableField(computeProfileStoryProgressText())
 
   fun onHeaderClicked() {
     routeToProfileProgressListener.routeToProfileProgress(profile.get()!!.id.internalId)
   }
 
   fun setOngoingTopicProgress(ongoingTopicCount: Int) {
     this.ongoingTopicCount = ongoingTopicCount
-    profileProgressText.set(computeProfileProgressText())
+    profileTopicProgressText.set(computeProfileTopicProgressText())
   }
 
   fun setCompletedStoryProgress(completedStoryCount: Int) {
     this.completedStoryCount = completedStoryCount
-    profileProgressText.set(computeProfileProgressText())
+    profileStoryProgressText.set(computeProfileStoryProgressText())
   }
 
-  private fun computeProfileProgressText(): String {
+  private fun computeProfileStoryProgressText(): String {
     // TODO(#3843): Either combine these strings into one or use separate views to display them.
```

Comment:
Please remove this TODO since it will cause a TODO check failure once the corresponding issue is closed